src/core/ucc_lib.c

```diff
@@ -67,9 +67,11 @@ static ucc_status_t ucc_cl_lib_init(const ucc_lib_params_t *user_params,
 {
     int                   n_cls = ucc_global_config.cl_framework.n_components;
     uint64_t              supported_coll_types = 0;
-    ucc_thread_mode_t     supported_tm         = UCC_THREAD_MULTIPLE;
+    ucc_thread_mode_t     highest_tm           = UCC_THREAD_SINGLE;
+    ucc_thread_mode_t     lowest_tm            = UCC_THREAD_MULTIPLE;
```

Comment:
This should be the opposite right? UCC_THREAD_SINGLE < UCC_THREAD_FUNNELED < UCC_THREAD_MULTIPLE ?

Reply:
yes, this is exactly how it is now. These are initialized values "to search for highest and lowest" modes among CLs. See the code below. I initialize highest to SINGLE and if anything lowest to MULTIPLE in order to search max/min values.